examples/nlp/mlm_and_finetune_with_bert.py

```diff
@@ -0,0 +1,290 @@
+
+    Returns:
+        tf.keras.layers.Layer: Return TextVectorization Keras Layer
+    """
+    vectorize_layer = TextVectorization(
+        max_tokens=vocab_size,
+        output_mode="int",
+        standardize=None,
+        output_sequence_length=max_seq,
+    )
+
+    vectorize_layer.adapt(texts)
+
+    vocab = vectorize_layer.get_vocabulary()
+    vocab = vocab[2 : vocab_size - len(special_tokens)] + ["[MASK]"]
+    vectorize_layer.set_vocabulary(vocab)
+    return vectorize_layer
+
+
+vectorize_layer = get_vectorize_layer(
+    data.text.values.tolist(),
+    flags.VOCAB_SIZE,
+    flags.MAX_LEN,
+    special_tokens=["[MASK]"],
+)
+
+# get mask token id for mlm
+mask_token_id = vectorize_layer(["[MASK]"]).numpy()[0][0]
+
+
+class IMDBReviewsDataGenerator(tf.keras.utils.Sequence):
+    """Generates batches of data
+
+    Args:
+        texts (list): List of String i.e input texts
+        labels (list): List of Labels i.e for classification model only
+        batch_size (int): batch size
+        vectorizer (TextVectorization): Keras TextVectorization layer
+        is_training (bool, optional): Is training dataset generator. Defaults to True.
+        model_type (str, optional): Used as a flag for mlm and classifier. Defaults to 'classification'.
+    """
+
+    def __init__(
+        self,
+        texts,
+        labels,
+        batch_size,
+        vectorizer,
+        is_training=True,
+        model_type="classification",
+    ):
+        self.texts = texts
+        self.labels = labels
+        self.batch_size = batch_size
+        self.vectorizer = vectorizer
+        self.is_training = is_training
+        self.model_type = model_type
+        self.indexes = np.arange(len(self.texts))
+
+    def __len__(self):
+        # Denotes the number of batches per epoch.
+        return len(self.texts) // self.batch_size
+
+    def __getitem__(self, idx):
+        # Retrieves the batch of index.
+        indexes = self.indexes[idx * self.batch_size : (idx + 1) * self.batch_size]
+        texts = self.texts[indexes]
+
+        encoded = self.vectorizer(texts).numpy()
+
+        if self.is_training and self.model_type == "mlm":
+            X_mlm, y_labels, sample_weights = self.prepare_mlm_input_and_labels(encoded)
+            return (X_mlm, y_labels, sample_weights)
+
+        elif self.is_training and self.model_type == "classification":
+            labels = np.array(self.labels[indexes], dtype="int32")
+            return (encoded, labels)
+
+        else:
+            return encoded
+
+    def prepare_mlm_input_and_labels(self, X):
+        # 15% BERT masking
+        inp_mask = np.random.rand(*X.shape) < 0.15
+        # do not mask special tokens
+        inp_mask[X <= 2] = False
+        # set targets to -1 by default, it means ignore
+        labels = -1 * np.ones(X.shape, dtype=int)
+        # set labels for masked tokens
+        labels[inp_mask] = X[inp_mask]
+
+        # prepare input
+        X_mlm = np.copy(X)
```

Comment:
Style nit: variable names should be lowercase. Please always use fully-spelled out variable names for better readability, instead of abbreviations or single letters.